src/cio_os.c

```diff
@@ -58,6 +60,8 @@ int cio_os_mkpath(const char *dir, mode_t mode)
         return 0;
     }
 
-    cio_os_mkpath(dirname(strdupa(dir)), mode);
+    dup_dir = strdup(dir);
```

Comment:
You need to check the return value here.
Otherwise, SIGSEGV brows out the stack.